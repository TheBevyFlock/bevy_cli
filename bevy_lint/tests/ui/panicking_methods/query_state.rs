//! This tests the `panicking_methods` lint, specifically when triggered on the `QueryState`
//! type.

#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::panicking_methods)]

use bevy::prelude::*;

#[derive(Component)]
struct Foo;

fn main() {
    let mut world = World::new();

    let mut query_state = QueryState::<&mut Foo>::new(&mut world);

    let _ = query_state.single(&world);
    //~^ ERROR:  called a `QueryState` method that can panic when a non-panicking alternative exists
    //~| HELP: use `query_state.get_single(&world)`

    let _ = QueryState::single(&mut query_state, &world);
    //~^ ERROR:  called a `QueryState` method that can panic when a non-panicking alternative exists
    //~| HELP: use `QueryState::get_single(&mut query_state, &world)`

    query_state.single_mut(&mut world);
    //~^ ERROR:  called a `QueryState` method that can panic when a non-panicking alternative exists
    //~| HELP: use `query_state.get_single_mut(&mut world)`

    QueryState::single_mut(&mut query_state, &mut world);
    //~^ ERROR:  called a `QueryState` method that can panic when a non-panicking alternative exists
    //~| HELP: use `QueryState::get_single_mut(&mut query_state, &mut world)`
}
