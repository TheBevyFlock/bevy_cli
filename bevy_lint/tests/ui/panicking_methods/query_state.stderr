error: called a `QueryState` method that can panic when a non-panicking alternative exists
  --> tests/ui/panicking_methods/query_state.rs:18:25
   |
18 |     let _ = query_state.single(&world);
   |                         ^^^^^^^^^^^^^^
   |
   = help: use `query_state.get_single(&world)` and handle the `Option` or `Result`
note: the lint level is defined here
  --> tests/ui/panicking_methods/query_state.rs:6:9
   |
6  | #![deny(bevy::panicking_query_methods)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: called a `QueryState` method that can panic when a non-panicking alternative exists
  --> tests/ui/panicking_methods/query_state.rs:22:17
   |
22 |     query_state.single_mut(&mut world);
   |                 ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `query_state.get_single_mut(&mut world)` and handle the `Option` or `Result`

error: aborting due to 2 previous errors

